deepmd/nvnmd/entrypoints/mapt.py

```diff
@@ -74,172 +76,354 @@ def __init__(
+                            a, b, c, d = coef[:,0], coef[:,1], coef[:, 2], coef[:, 3]
+                            dat_i[kk, :] = d + (c + (b + a * dxx_k) * dxx_k ) * dxx_k
+                        elif rank == 2:
+                            coef = np.reshape(coef, [nc, 2])
+                            a, b = coef[:,0], coef[:,1]
+                            dat_i[kk, :] = b + a * dxx_k
+                    dats.append(dat_i)
+                dic_val[key] = dats
+        return dic_val
+
+    def mapping2(self, x, dic_map, cfgs, rank=4):
+        """ Evaluate value by mapping table of numpy   
+        """
+        tf.reset_default_graph()
+        t_x = tf.placeholder(tf.float64, [None, 1], 't_x')
+        t_table = tf.placeholder(tf.float64, [None, None], 't_table')
+        t_table_grad = tf.placeholder(tf.float64, [None, None], 't_table_grad')
+        t_table_info = tf.placeholder(tf.float64, [None], 't_table_info')
+        t_y = op_module.map_flt_nvnmd(t_x, t_table, t_table_grad, t_table_info)
+        sess = get_sess()
+        #
+        n = len(x)
+        dic_val = {}
+        for key in dic_map.keys():
+            val = dic_map[key]
+            if isinstance(val, list):
+                dats = []
+                for ii in range(len(val)):
+                    val_i = val[ii]
+                    feed_dict = {
+                        t_x : x,
+                        t_table : val_i,
+                        t_table_grad : val_i * 0.0,
+                        t_table_info : np.reshape(np.array(cfgs), [-1])
+                    }
+                    dat_i = run_sess(sess, t_y, feed_dict=feed_dict)
+                    dat_i = np.reshape(dat_i, [n, -1])
+                    dats.append(dat_i)
+                dic_val[key] = dats
+        return dic_val
+
+    def plot_lines(self, x, dic1, dic2=None):
+        """ Plot lines to see accuracy
+        """
+        for key in dic1.keys():
+            val1 = dic1[key]
+            if dic2 is None:
+                val2 = dic1[key]
+            else:
+                val2 = dic2[key]
+            #
+            if isinstance(val1, list):
+                for ii in range(len(val1)):
+                    val1_i = val1[ii]
+                    val2_i = val2[ii]
+                    nc = np.shape(val1_i)[1]
+                    # For debug
+                    # plt.figure
+                    # for jj in range(nc):
+                    #     c1 = 'r'
+                    #     c2 = 'k'
+                    #     plt.plot(x, val1_i[:,jj], '-', color=c1, linewidth=6)
+                    #     plt.plot(x, val2_i[:,jj], '-', color=c2, linewidth=2)
+                    # plt.title(key)
+                    # plt.grid()
+                    # plt.savefig('mapt_'+key+'-'+str(ii)+'.png')
+                    # plt.close()
+
+    def build_map_coef(self, cfgs, x, ys, grads, grad_grads, Nr, Nc, rank=4):
+        """ Build mapping table coefficient
```

Comment:
```suggestion
        r""" Build mapping table coefficient
```

Reply:
Thank you very much. We have made changes.